### PR TITLE
Agoric cli improvements

### PR DESCRIPTION
**Pull request by michaelfig**
Some changes to address things that @dtribble found wrong:

* Ensure that `yarn link-cli /some/path/to/agoric` check that `/some/path/to` is in the user's `$PATH`.
* Reenable `package.json` substitutions.

### CODE REVIEW

File: packages/agoric-cli/lib/init.js [hunk 1]
```diff
@@ -62,6 +62,29 @@ export default async function initMain(_progname, rawArgs, priv, _opts) {
...
+    if (!pkg.name || !pkg.name.startsWith(topLevelName)) {
+      throw Error(
+        `${path}: "name" must start with ${JSON.stringify(topLevelName)}`,
+      );
+    }
+    pkg.name = `${DIR}${pkg.name.substr(topLevelName.length)}`;
+    const json = JSON.stringify(pkg, undefined, 2);
+
+    // eslint-disable-next-line no-await-in-loop
+    await fs.writeFile(path, json);
```
Comment (FUDCo) [inline packages/agoric-cli/lib/init.js:85]: Is there some new Node feature I was unaware of that converts Node's callback-based APIs into promise-based APIs without using util.promisify or its ilk?  I.e., is this `await fs.writeFile` a mistake or am I just not up to date on the latest hotness?

File: scripts/link-cli.cjs [hunk 2]
```diff
@@ -4,8 +4,31 @@ const fs = require('fs');
 const path = require('path');
 
 try {
-  const script = process.argv[2] || '/usr/local/bin/agoric';
+  const script = process.argv[2] || `${process.env.HOME || '/usr/local'}/bin/agoric`;
```
Comment (FUDCo) [inline scripts/link-cli.cjs:7]: Is there a presumption here that a user's personal local executables are in a directory named `bin`?  Because mine aren't.

**Review by FUDCo — APPROVED** [review]
I flagged a couple of things, but I'm not sure if they're things that need to be corrected or just stuff that I've misunderstood.  Approving on the presumption that it's me, but check to be sure.

File: packages/agoric-cli/lib/init.js (hunk 1 above)
Comment (michaelfig) [reply packages/agoric-cli/lib/init.js:85]: `fs` is actually `fs.promises`, which does have promise-based APIs.  Perhaps we should rename the power from `fs` to `fsp` or something.

File: scripts/link-cli.cjs (hunk 2 above)
Comment (michaelfig) [reply scripts/link-cli.cjs:7]: Not really a presumption, just a default.  You can override the default with an argument (which power users probably will do).

**Pull request merged**